### DISCUSSION TITLE
Anyone tinkered with emlearn on an esp32?

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#12999] [Libraries & Drivers]
https://github.com/emlearn/emlearn-micropython claims to offer upython based TinyML for the esp32. Anyone tried it? 

I'm a complete novice when it comes to ML would 'low-complexity' exclude photo classification?

What does 'windowed with window size 8' in the processor architecture table mean?

**Comment by jimmo** [comment]
I haven't tried it but I've talked to the author about it. It's a great approach to providing this sort of functionality, and for the types of problems that the provided classifiers work for, it's a great option.

> I'm a complete novice when it comes to ML would 'low-complexity' exclude photo classification?

It supports RandomForest/DecisionTree models and K-Nearest Neighbors. It depends what you mean by "photo classification". (For example it includes the standard machine-learning handwriting digit-recognition demo as an example).

> What does 'windowed with window size 8' in the processor architecture table mean?

It means "esp32".

**Comment by jonnor** [comment]
Hi. emlearn maintainer here. I had not seen this post.

The goal of emlearn-micropython is to make TinyML accessible for those that know Python and basics of ML. However, TinyML must still be considered a bit more tricky than "plain" ML (without the resource constraints, running on the same machine as one is developing). I would always recommend doing a couple of projects on PC first, just to get a feel for it. The training will need to happen there anyway, so one needs to get comfortable with that side in any case :)

Just some weeks ago I added support for Convolutional Neural Networks - the best model for image classification. It still needs some testing and documentation, but over next months that should be in place :)

**Comment by peterhinch** [comment]
The people at [OpenMV](https://openmv.io/) are doing image classification with MicroPython.
